src/Rings/mpoly-localizations.jl

```diff
@@ -382,6 +382,77 @@ function rand(S::MPolyProductOfMultSets, v1::UnitRange{Int}, v2::UnitRange{Int},
   return prod([rand(s, v1, v2, v3) for s in sets(S)])::elem_type(ambient_ring(S))
 end
 
+########################################################################
+# Localization associated to a monomial ordering                       #
+########################################################################
+
+@Markdown.doc """
+    MPolyLeadingMonOne{
+        BaseRingType,
+        BaseRingElemType,
+        RingType,
+        RingElemType
+      } <: AbsMPolyMultSet{
+        BaseRingType,
+        BaseRingElemType,
+        RingType,
+        RingElemType
+      }
+
+The set `S = { a in R : leading_monomial(a, ord) = 1 }` for a fixed
+monomial ordering `ord`.
+"""
+mutable struct MPolyLeadingMonOne{
+    BaseRingType,
+    BaseRingElemType,
+    RingType,
+    RingElemType
+  } <: AbsMPolyMultSet{
+    BaseRingType,
+    BaseRingElemType,
+    RingType,
+    RingElemType
+  }
+
+  R::RingType # the parent ring
+  ord::MonomialOrdering
```

Comment:
Again, this is not a concrete type, is it? We should see whether this leads to a performance issue, but probably not.